### PR TITLE
documentation update and typehints for visualization 

### PR DESCRIPTION
**Pull request by plakrisenko**

### CODE REVIEW

File: petab/visualize/plot_data_and_simulation.py [hunk 1]
```diff
@@ -219,19 +222,21 @@ def plot_petab_problem(petab_problem: problem.Problem,
 
 def plot_measurements_by_observable(data_file_path: str,
                                     condition_file_path: str,
-                                    plotted_noise: str = MEAN_AND_SD):
+                                    plotted_noise: Optional[str] = MEAN_AND_SD
+                                    ) -> Optional[Union[Dict[str, plt.Subplot],
+                                                  'np.ndarray[plt.Subplot]']]:
```
Comment (dweindl) [inline petab/visualize/plot_data_and_simulation.py]: Codacy Issue found: [Wrong continued indentation (add 6 spaces).](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5135473)

File: doc/documentation_data_format.md [hunk 2]
```diff
@@ -436,7 +436,7 @@ provided) inside the measurement table.
 Expected to have the following columns in any (but preferably this)
 order:
 
-| plotId | [plotName] | plotTypeSimulation | plotTypeData | datasetId | ...
+| plotId | [plotName] | [plotTypeSimulation] | [plotTypeData] | [datasetId] | ...
```
Comment (erikadudki) [inline doc/documentation_data_format.md]: so far, I think datasetId and xValues are mandatory.
Question: If we want to change that, that would mean the only mandatory column would be plotId, does this make sense? What should then be plotted if just plotId is provided?

File: doc/documentation_data_format.md [hunk 3]
```diff
@@ -463,62 +463,66 @@ order:
...
 
-- `yLabel` [STRING]
+- `yLabel` [STRING, OPTIONAL]
 
   Label for the y-axis.
 
-- `yScale` [STRING]
+- `yScale` [STRING, OPTIONAL]
 
   Scale of the observable, can be `lin`, `log`, or `log10`.
```
Comment (erikadudki) [inline doc/documentation_data_format.md]: Adding: Default is 'lin'?

Comment (plakrisenko) [inline doc/documentation_data_format.md]:
```suggestion
  Scale of the observable, can be `lin`, `log`, or `log10`. Default is `lin`.
```

File: doc/documentation_data_format.md (hunk 2 above)
Comment (plakrisenko) [reply doc/documentation_data_format.md]: Well, in a way it still makes sense, as it allows to specify values different from defaults only for a subset of optional columns. However, if there is only one column in vis_spec, maybe we should show some message.

Comment (erikadudki) [reply doc/documentation_data_format.md]: okay, sounds good :+1:

**Pull request merged**